File: src/main/java/com/dev/farmmanager/service/item/ItemService.java
package com.dev.farmmanager.service.item;

import com.dev.farmmanager.domain.entity.Item;
import com.dev.farmmanager.domain.payload.item.ItemPayload;
import lombok.NonNull;

import java.util.List;
import java.util.Optional;

public interface ItemService {

    List<Item> findAll();

    Optional<Item> getById(@NonNull Integer id);

    Item create(@NonNull Integer userId, @NonNull ItemPayload payload);

    Item update(@NonNull Integer id, @NonNull ItemPayload payload);

    void delete(@NonNull Integer id);
}

File: frontend/src/services/item.service.ts
import http from '@/api/http'
import type { ItemDto, ItemPayload } from '@/types/item'

export const itemService = {
  findAll: () =>
    http.get<ItemDto[]>('/farm/item').then((r) => r.data),

  getById: (id: number) =>
    http.get<ItemDto>(`/farm/item/${id}`).then((r) => r.data),

  create: (payload: ItemPayload) =>
    http.post<ItemDto>('/farm/item', payload).then((r) => r.data),

  update: (id: number, payload: ItemPayload) =>
    http.put<ItemDto>(`/farm/item/${id}`, payload).then((r) => r.data),

  delete: (id: number) =>
    http.delete(`/farm/item/${id}`),
}
